eclair-core/src/main/scala/fr/acinq/eclair/payment/Relayer.scala

```diff
@@ -49,74 +52,58 @@ case class ForwardFailMalformed(fail: UpdateFailMalformedHtlc, to: Origin, htlc:
+  context.system.eventStream.subscribe(self, classOf[AvailableBalanceChanged])
 
   val commandBuffer = context.actorOf(Props(new CommandBuffer(nodeParams, register)))
 
-  override def receive: Receive = main(Map())
-
-  def main(channelUpdates: Map[ShortChannelId, ChannelUpdate]): Receive = {
-
-    case LocalChannelUpdate(_, channelId, shortChannelId, remoteNodeId, _, channelUpdate) =>
-      log.debug(s"updating channel_update for channelId=$channelId shortChannelId=$shortChannelId remoteNodeId=$remoteNodeId channelUpdate=$channelUpdate ")
-      context become main(channelUpdates + (channelUpdate.shortChannelId -> channelUpdate))
-
-    case LocalChannelDown(_, channelId, shortChannelId, _) =>
-      log.debug(s"removed local channel_update for channelId=$channelId shortChannelId=$shortChannelId")
-      context become main(channelUpdates - shortChannelId)
-
-    case ForwardAdd(add) =>
-      log.debug(s"received forwarding request for htlc #${add.id} paymentHash=${add.paymentHash} from channelId=${add.channelId} ")
-      Sphinx.parsePacket(nodeParams.privateKey, add.paymentHash, add.onionRoutingPacket)
-        .flatMap {
-          case Sphinx.ParsedPacket(payload, nextPacket, sharedSecret) =>
-            LightningMessageCodecs.perHopPayloadCodec.decode(BitVector(payload.data)) match {
-              case Attempt.Successful(DecodeResult(perHopPayload, _)) => Success((perHopPayload, nextPacket, sharedSecret))
-              case Attempt.Failure(cause) => Failure(new RuntimeException(cause.messageWithContext))
-            }
-        } match {
-        case Success((perHopPayload, nextPacket, _)) if nextPacket.isLastPacket =>
-          val cmd = perHopPayload match {
-            case PerHopPayload(_, finalAmountToForward, _) if finalAmountToForward > add.amountMsat =>
-              Left(CMD_FAIL_HTLC(add.id, Right(FinalIncorrectHtlcAmount(add.amountMsat)), commit = true))
-            case PerHopPayload(_, _, finalOutgoingCltvValue) if finalOutgoingCltvValue != add.expiry =>
-              Left(CMD_FAIL_HTLC(add.id, Right(FinalIncorrectCltvExpiry(add.expiry)), commit = true))
-            case _ =>
-              Right(add)
-          }
-          cmd match {
+  override def receive: Receive = main(Map.empty, new mutable.HashMap[PublicKey, mutable.Set[ShortChannelId]] with mutable.MultiMap[PublicKey, ShortChannelId])
+
+  def main(channelUpdates: Map[ShortChannelId, OutgoingChannel], node2channels: mutable.HashMap[PublicKey, mutable.Set[ShortChannelId]] with mutable.MultiMap[PublicKey, ShortChannelId]): Receive = {
+
+    case LocalChannelUpdate(_, channelId, shortChannelId, remoteNodeId, _, channelUpdate, commitments) =>
+      log.debug(s"updating local channel info for channelId=$channelId shortChannelId=$shortChannelId remoteNodeId=$remoteNodeId channelUpdate={} commitments={}", channelUpdate, commitments)
+      val availableLocalBalance = commitments.remoteCommit.spec.toRemoteMsat
```

Comment:
nitpick: there should be a comment here on why we use the to_remote field in our view of their remote tx

Reply:
fixed